openslides_backend/action/motion_category/number_motions.py

```diff
@@ -0,0 +1,225 @@
+        self.helper_affected_categories(queue[1:], aff_categories)
+
+    def sort_category_children(self, child_ids: List[int]) -> List[int]:
+        """Sort the categories by weight. Important for the helper_affected_categories."""
+        weighted_child_ids = []
+        for id_ in child_ids:
+            category = self.mem_categories.get(id_, {})
+            weighted_child_ids.append((category.get("weight", 10000), id_))
+        weighted_child_ids.sort()
+        return [id_ for (_, id_) in weighted_child_ids]
+
+    def get_affected_motions(self, category_ids: List[int]) -> List[int]:
+        """Get the affected motions from the categories"""
+        affected_motions = []
+        for category_id in category_ids:
+            category = self.mem_categories.get(category_id, {})
+            affected_motions.extend(self.sort_motions(category.get("motion_ids", [])))
+        return affected_motions
+
+    def sort_motions(self, ids_: List[int]) -> List[int]:
+        """Sort motions by category_weight."""
+        weighted_motion_ids = []
+        for id_ in ids_:
+            motion = self.mem_motions.get(id_, {})
+            weighted_motion_ids.append((motion.get("category_weight", 10000), id_))
+        weighted_motion_ids.sort()
+        return [id_ for (_, id_) in weighted_motion_ids]
+
+    def get_number(self, motion_id: int, number_value_map: Dict[int, int]) -> str:
+        """Get number, uses the number_value_map, two main cases."""
+        lead_motion_id = self.get_lead_motion_id(motion_id)
+        if lead_motion_id:
+            lead_number = self.get_number(lead_motion_id, number_value_map)
+            meeting_id = self.get_meeting_id(motion_id)
+            if meeting_id:
+                meeting = self.mem_meetings.get(meeting_id, {})
+            else:
+                meeting = {
+                    "motions_number_with_blank": False,
+                    "motions_number_min_digits": 0,
+                    "motions_amendments_prefix": "",
+                }
+            blank = " " if meeting.get("motions_number_with_blank") else ""
+            amendments_prefix = meeting.get("motions_amendments_prefix", "")
+            number_value_str = str(number_value_map[motion_id]).rjust(
+                meeting.get("motions_number_min_digits", 0), "0"
+            )
+            number = f"{lead_number}{blank}{amendments_prefix}{blank}{number_value_str}"
+        else:
+            motion = self.mem_motions.get(motion_id, {})
+            if motion.get("category_id"):
+                prefix = self.get_prefix(motion["category_id"])
+            else:
+                prefix = ""
+            meeting_id = self.get_meeting_id(motion_id)
+            if meeting_id:
+                meeting = self.mem_meetings.get(meeting_id, {})
+            else:
+                meeting = {
+                    "motions_number_with_blank": False,
+                    "motions_number_min_digits": 0,
+                }
```

Comment:
same as above, meeting_id must be set for all models

Reply:
Same here.